src/DocumentFragment.js

```diff
@@ -0,0 +1,71 @@
+  // ### DocumentFragment Shim
+  //
+  // A representation of a document fragment that can be normally generated
+  // by calling `document.createDocumentFragment`. Since the document fragment
+  // is itself a node, it inherits all properties of a node.
+module.exports = (function(){
+  "use strict";
+
+  var Node = require('./Node');
+  var voidHtmlElements = ["area", "base", "br", "col", "command", "embed", "hr",
+                          "img", "input", "keygen", "link", "meta", "param",
+                          "source", "track", "wbr"];
+
+  function assertExists(name, value) {
+    if(!value || value === "") {
+      throw new Error("Expected a value for "  + name);
+    }
+
+    return true;
+  }
+
+  function initNode(node, type, name) {
+    Object.defineProperties(node, {
```

Comment:
There is no reason for this. Give nodes their own type and name, and define these properties on the Node.prototype instead.